### PR TITLE
LLM: support `optimize_model=True` for Mixtral model

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Support `optimize_model=True` for Mixtral model:
- [ ] attention forward
- [x] RMSNorm

### 4. How to test?
- [ ] Unit test
- [x] Local test

### CODE REVIEW

File: python/llm/src/bigdl/llm/transformers/convert.py [hunk 1]
```diff
@@ -616,6 +616,15 @@ def _optimize_post(model, lightweight_bmm=False):
         convert_forward(model,
                         module.MistralRMSNorm,
                         llama_rms_norm_forward)
+    elif model.config.model_type == "mixtral":
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/convert.py:619]: Also check `model.config.architectures` for mistral and mixtral (to support DiscoResearch models)?

Comment (plusbang) [reply python/llm/src/bigdl/llm/transformers/convert.py]: > Also check `model.config.architectures` for mistral and mixtral (to support DiscoResearch models)?

Have added related check.

File: python/llm/src/bigdl/llm/transformers/convert.py [hunk 2]
```diff
@@ -616,6 +616,24 @@ def _optimize_post(model, lightweight_bmm=False):
...
+            convert_forward(model,
+                            module.MixtralRMSNorm,
+                            llama_rms_norm_forward)
+        elif model.config.model_type == "mistral":
+            # For DiscoResearch/mixtral-7b-8expert
+            modeling_module_name = model.__class__.__module__
+            module = importlib.import_module(modeling_module_name)
+            convert_forward(model,
+                            module.MistralRMSNorm,
+                            llama_rms_norm_forward)
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/transformers/convert.py:636]:
```python
if model.config.model_type == "mixtral":
    ...
elif model.config.model_type == "mistral":
   if model.config.architectures is not None and model.config.architectures[0] == "MixtralForCausalLM":
      ...
   else
      ...
```

**Review by jason-dai — APPROVED** [review]
LGTM

**Pull request merged**